tests/packet_tests.cpp

```diff
@@ -105,19 +103,87 @@ BOOST_AUTO_TEST_CASE(PacketBufferConstructor)
     BOOST_CHECK(types.count(attrs[4]->type()) == 1);
     BOOST_CHECK(types.count(attrs[5]->type()) == 1);
 
+    std::vector<uint8_t> values = p.makeSendBuffer("secret");
+
+    BOOST_REQUIRE_EQUAL(values.size(), 92);
+
+    std::vector<uint8_t> expected({0x01, 0xd0, 0x00, 0x5c, 0x1a, 0x40, 0x43, 0xc6, 0x41, 0x0a, 0x08, 0x31, 0x12, 0x16, 0x80, 0x2c, 0x3e, 0x83, 0x12, 0x45, 0x01, 0x06, 0x74, 0x65, 0x73, 0x74, 0x02, 0x12, 0x8c, 0x06, 0xc8, 0x23, 0x55, 0xba, 0x0d, 0xd6, 0x15, 0x1c, 0xbf, 0x9d, 0xd8, 0x1a, 0x4d, 0x87, 0x04, 0x06, 0x7f, 0x00, 0x00, 0x01, 0x05, 0x06, 0x00, 0x00, 0x00, 0x01, 0x50, 0x12, 0xf3, 0xe0, 0x00, 0xe7, 0x7d, 0xeb, 0x51, 0xeb, 0x81, 0x5d, 0x52, 0x37, 0x3d, 0x06, 0xb7, 0x1b, 0x07, 0x06, 0x00, 0x00, 0x00, 0x01, 0x1a, 0x0c, 0x00, 0x00, 0x00, 0xab, 0x01, 0x06, 0x00, 0x00, 0x00, 0x03});
```

Comment:
I think this is the same as `d` and you should use the `d` in the comparison below.